File: hacker-rank/thirty-days-of-code/day-08/Solution.java
//Complete this code or write your own from scratch
import java.util.*;
import java.io.*;

class Solution{
  public static void main(String []arg){
    HashMap<String, Integer> phoneBook = new HashMap<String, Integer>();

    Scanner in = new Scanner(System.in);
    int n = in.nextInt();

    for(int i = 0; i < n; i++){
      String name = in.next();
      int phoneNumber = in.nextInt();
      // Write code here
      phoneBook.put(name, phoneNumber);
    }

    while(in.hasNext()){
      String queryName = in.next();
      String message = "Not found";
      // Write code here
      if (phoneBook.get(queryName) != null) {
        int phoneNumber = phoneBook.get(queryName);
        message = (queryName + "=" + phoneNumber);
      }
      System.out.println(message);
    }
    in.close();
  }
}

File: hacker-rank/thirty-days-of-code/day-08/solution.js
function processData(input) {
  //Enter your code here
  let inputArray = input.split("\n")
  let indexOfFirstQuery = Number(inputArray[0]) + 1
  let entries = inputArray.slice(1, indexOfFirstQuery)
  let queries = inputArray.slice(indexOfFirstQuery)

  let phoneBook = new Map()
  entries.forEach(inputLine => {
    let inputLineItems = inputLine.split(" ")
    let name = inputLineItems[0]
    let phoneNumber = inputLineItems[1]
    phoneBook.set(name, phoneNumber)
  }); 

  queries.forEach((name) => {
    let number = phoneBook.get(name)
    let message = number ? `${name}=${number}` : "Not found"
    console.log(message)
  })
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
 processData(_input);
});
